MultiTable/test/RealtimeChangeHandler.js

```diff
@@ -14,11 +14,73 @@ describe('./test/RealtimeChangeHandler.js >', () => {
     const data = new SubjectGroups(sgArgs);
     const chg = { 'sample.add': { name: 'a|b', aspect: { name: 'b'} } };
     RealtimeChangeHandler.handle(chg, data);
-    expect(data instanceof SubjectGroups).to.equal.true;
+    expect(data instanceof SubjectGroups).to.be.true;
     expect(data).to.have.property('rootSubject', 'a');
     expect(data.map).to.have.property('a');
   });
 
+  describe('handle: updateSample >', () => {
+    it('updateSample should be called for a sample update event', () => {
+      const _sgArgs = { absolutePath: 'Fellowship', name: 'Fellowship'}
+      const data = new SubjectGroups(_sgArgs);
+      const sample = { name: 'Fellowship|View', status: 'OK',
+        aspect: { name: 'View'} };
+      let chg = { 'sample.add': sample };
+      RealtimeChangeHandler.handle(chg, data);
+      const updatedSample = { name: 'Fellowship|View', status: 'Critical',
+        aspect: { name: 'View'} };
+      chg = { 'sample.update': { new : updatedSample } };
+      RealtimeChangeHandler.handle(chg, data);
+
+      /*
+       * make sure that the samples instance object of the subjectGroup is
+       * updated
+       */
+      expect(data.map.fellowship.samples['fellowship|view'])
+        .to.eq(updatedSample);
+      expect(Object.keys(data.map.fellowship.samples)).to.have.lengthOf(1);
+
+      /*
+       * make sure that the subjects instance object of the subjectGroup has
+       * the updated sample
+       */
+      expect(data.map.fellowship.subjects.fellowship.samples)
+        .to.have.lengthOf(1);
+      expect(data.map.fellowship.subjects.fellowship.samples[0]).to
+        .eql(updatedSample);
+     });
+
+    it('updateSample should add sample to the subject array if not ' +
```

Comment:
Would be good to also have a test that replicates the exact situation, where a subject is deleted then added back.

Reply:
From the tests done today, it was clear that this fix does not solve the problem. I was able to narrow down the problem to the addSubject method and I am creating a bug describing the exact scenario on which the pods disappear. We can write the test as a part of that bug.